docs/celsius-converter-exception/Snippets.re

```diff
@@ -0,0 +1,234 @@
+module CelsiusConverterOriginal = {
+  // #region celsius-converter-v1
+  let convert = celsius => 9.0 /. 5.0 *. celsius +. 32.0;
+
+  [@react.component]
+  let make = () => {
+    let (celsius, setCelsius) = React.useState(() => "");
+
+    <div>
+      <input
+        value=celsius
+        onChange={evt => {
+          let newCelsius = ReactEvent.Form.target(evt)##value;
+          setCelsius(_ => newCelsius);
+        }}
+      />
+      {React.string({js|°C = |js})}
+      {React.string({js|?°F|js})}
+    </div>;
+  };
+  // #endregion celsius-converter-v1
+};
+
+module CelsiusConverterGetValueFromEvent = {
+  // #region get-value-from-event
+  let getValueFromEvent = (evt): string =>
+    ReactEvent.Form.target(evt)##value;
+
+  let convert = celsius => 9.0 /. 5.0 *. celsius +. 32.0;
+
+  [@react.component]
+  let make = () => {
+    let (celsius, setCelsius) = React.useState(() => "");
+
+    <div>
+      <input
+        value=celsius
+        onChange={evt => {
+          let newCelsius = getValueFromEvent(evt);
+          setCelsius(_ => newCelsius);
+        }}
+      />
+      {React.string({js|°C = |js})}
+      {React.string({js|?°F|js})}
+    </div>;
+  };
+  // #endregion get-value-from-event
+};
+
+let _ = {
+  let celsius = "";
+  let convert = x => x;
+  // #region pipes
+  <div>
+    <input /* ... */ />
```

Comment:
Fortunately `/* */` comments inside a component don't get shuffled around in JSX expressions